### PR TITLE
Import opm core

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -92,6 +92,26 @@ macro (sources_hook)
       ${PROJECT_SOURCE_DIR}/tutorials/tutorial4.cpp
       )
   endif()
+
+  # these solvers are only compiled in if their dependency is found
+  if (NOT dune-istl_FOUND)
+    list (REMOVE_ITEM opm-simulators_SOURCES
+      ${PROJECT_SOURCE_DIR}/${opm-core_DIR}/core/linalg/LinearSolverIstl.cpp
```
Comment (andlaus) [inline CMakeLists.txt]: this seems to be wrong? (OTOH first adding  these files unconditionally just to remove them later seems to be quite sketchy anyway.)

Comment (andlaus) [reply CMakeLists.txt]: come think about it, it might be correct. **Iff** opm-core_DIR is `opm/core`. is it?

Comment (akva2) [reply CMakeLists.txt]: yes i missed the dune.module removal, it has been added after your review.

and yes, it works due to the second comment there, but it is not very pretty so i will change it to literal opm. i agree that the add / removal is rather meh but it's how things have been done earlier so didn't want to change that now.

Comment (andlaus) [reply CMakeLists.txt]: > and yes, it works due to the second comment there, but it is not very pretty so i will change it to literal opm.

:+1:

> i agree that the add / removal is rather meh but it's how things have been done earlier so didn't want to change that now.

I completely agree. I did not mean this to be an issue introduced by the PR.

**Pull request merged**